tests/unit/_experimental/test_plots.py

```diff
@@ -0,0 +1,1229 @@
+            ],
+            CurvePlot(
+                title="F1-Score vs. Confidence Threshold",
+                x_label="Confidence Threshold",
+                y_label="F1-Score",
+                curves=[
+                    Curve(x=[1.0], y=[0.0], label="zeros, but b is confused with a"),
+                ],
+                x_config=None,
+                y_config=None,
+            ),
+            CurvePlot(
+                title="Precision vs. Recall",
+                x_label="Recall",
+                y_label="Precision",
+                curves=[
+                    Curve(x=[0.0], y=[0.0], label="zeros, but b is confused with a"),
+                ],
+                x_config=None,
+                y_config=None,
+            ),
+        ),
+        (
+            "zeros, but a is confused with b",
+            [
+                MulticlassInferenceMatches(
+                    matched=[],
+                    unmatched_gt=[
+                        (LabeledBoundingBox((1, 1), (2, 2), "a"), None),
+                        (LabeledBoundingBox((1, 1), (2, 2), "b"), ScoredLabeledBoundingBox((1, 1), (2, 2), "a", 0)),
+                    ],
+                    unmatched_inf=[],
+                ),
+            ],
+            CurvePlot(
+                title="F1-Score vs. Confidence Threshold",
+                x_label="Confidence Threshold",
+                y_label="F1-Score",
+                curves=[
+                    Curve(x=[1.0], y=[0.0], label="zeros, but a is confused with b"),
+                ],
+                x_config=None,
+                y_config=None,
+            ),
+            CurvePlot(
+                title="Precision vs. Recall",
+                x_label="Recall",
+                y_label="Precision",
+                curves=[
+                    Curve(x=[0.0], y=[0.0], label="zeros, but a is confused with b"),
+                ],
+                x_config=None,
+                y_config=None,
+            ),
+        ),
+        (
+            "no confusion, one TP per label",
+            [
+                MulticlassInferenceMatches(
+                    matched=[
+                        (LabeledBoundingBox((3, 3), (4, 4), "b"), ScoredLabeledBoundingBox((3, 3), (4, 4), "b", 0)),
+                        (LabeledBoundingBox((5, 5), (6, 6), "a"), ScoredLabeledBoundingBox((5, 5), (6, 6), "a", 0)),
+                    ],
+                    unmatched_gt=[
+                        (LabeledBoundingBox((1, 1), (2, 2), "b"), None),
+                        (LabeledBoundingBox((1, 1), (2, 2), "a"), None),
+                    ],
+                    unmatched_inf=[],
+                ),
+            ],
+            CurvePlot(
+                title="F1-Score vs. Confidence Threshold",
+                x_label="Confidence Threshold",
+                y_label="F1-Score",
+                curves=[
+                    Curve(x=[0.0, 0.0, 0.0], y=[0.0, 0.0, 0.0], label="no confusion, one TP per label"),
```

Comment:
nit: probably doesn't really matter, but I think `np.linspace(0, 0, 4)[:-1]` results in duplicate thresholds of `0` — this is why we see `(4-1)` `(0, 0)` points.  Should we drop dup threshold?

Reply:
It seems like sklearn uses only the unique thresholds.  [sklearn link](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_recall_curve.html)
<img width="802" alt="image" src="https://github.com/kolenaIO/kolena/assets/17771952/bf81fcab-2352-4b0b-9f02-c3927914c93a">